docs/proposals/sig-device-iot/6312-device-anomaly-detection-framework/README.md

```diff
@@ -0,0 +1,344 @@
+- **Degradation strategy**: When the detector times out or becomes unavailable, it defaults to pass-through and logs the event.
+
+You can configure the related parameters for DetectorRouterHook in the `cloudcore` configuration file as shown below:
+
+```yaml
+deviceController:
+  buffer:
+    deviceEvent: 1
+    deviceModelEvent: 1
+    updateDeviceStatus: 1024
+  enable: true
+  load:
+    updateDeviceStatusWorkers: 1
+
+  # ===== New: Detector Router Hook Configuration =====
+  detectorRouterHooks:
+    - name: "temp-detector"              # Detector name
+      enable: true
+      filter:
+        namespace: "factory-a"
+        deviceLabels:
+          model: "pump-x"
+        properties:
+          - "temp"
+          - "pressure"
+      detectorEndpoint: "grpc://detector-temp.kubeedge.svc.cluster.local:50051"
+      timeout: 500ms                     # Detector inference timeout
+      retry:
+        maxRetries: 2
+        backoff: 200ms
+      tls:
+        enable: true
+        caFile: "/etc/kubeedge/certs/ca.crt"
+        certFile: "/etc/kubeedge/certs/client.crt"
+        keyFile: "/etc/kubeedge/certs/client.key"
+
+    - name: "statuscode-detector"        # Another detector
+      enable: true
+      filter:
+        namespace: "factory-b"
+        properties:
+          - "status_code"
+      detectorEndpoint: "grpc://detector-status.kubeedge.svc.cluster.local:50051"
+      timeout: 500ms
+      retry:
+        maxRetries: 2
+        backoff: 200ms
+      tls:
+        enable: false
+```
+
+### 4.4 AnomalyDetector
+
+**Location**: Independent cloud-side detection service, can be deployed in multiple instances.
+
+**Function**:  
+- Receives device state data from **DetectorRouterHook** via gRPC bidirectional streams.  
+- Calls built-in detection algorithms or loaded models for real-time inference.  
+- Returns `{anomaly mark, anomaly type, score, reason, model version}` to **DeviceController**.  
+- Optionally writes detection results to its own database for statistics and analysis.
+
+**Detector Console (Submodule)**:  
+- Provides real-time detection results, including device state curves, anomaly point markers, and detection hit rate statistics.  
+- Supports model version management and switching (with security controls).
+
+**Characteristics**:  
+- **Algorithm plug-in**: Supports interfaces for rules, statistics, machine learning, deep learning, and other algorithms.  
+- **Model hot loading**: Supports switching models without interrupting services.  
+- **Horizontal scaling**: Can scale according to load, supporting load balancing or device sharding.
+
+### 4.5 Result Writing and Alarm Module
+
+**Location**: Integrated within **DeviceController**.
+
+**Function**:  
+- Updates detection results to the corresponding `status.anomaly` field of the **Device CRD**, including the most recent detection state, score, type, and timestamp.  
```

Comment:
Perhaps `status.extensions`(map[string]any) is more appropriate, and it would be suitable for more scenes.